document-store/src/main/java/org/hypertrace/core/documentstore/mongo/parser/MongoIdentifierPrefixingSelectingExpressionParser.java

```diff
@@ -0,0 +1,22 @@
+package org.hypertrace.core.documentstore.mongo.parser;
+
+import static org.hypertrace.core.documentstore.mongo.MongoUtils.PREFIX;
+
+import java.util.Optional;
+import org.hypertrace.core.documentstore.expression.impl.IdentifierExpression;
+
+final class MongoIdentifierPrefixingSelectingExpressionParser
+    extends MongoSelectingExpressionParser {
+
+  MongoIdentifierPrefixingSelectingExpressionParser(
+      final MongoSelectingExpressionParser baseParser) {
+    super(baseParser);
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public String visit(final IdentifierExpression expression) {
+    assert baseParser != null;
```

Comment:
In the parent class, if `baseParse` is null, we are throwing `throw getUnsupportedOperationException(expression)`, and here we are asserting, what's the difference?

Reply:
Got rid with NoOpParser.